src/string.zig

```diff
@@ -189,6 +189,27 @@ pub const String = extern struct {
         return this;
     }
 
+    /// Convert dead strings into `error.OutOfMemory`.
```

Comment:
this is very risky to generalize because many apis that cross zig/c++ boundary use dead strings to also signify JSError and knowing which one is which is context dependent

Reply:
what do you prefer we do instead? should this be removed entirely?

Reply:
yeah better to do in-place at the callsite of the zig cpp call

Reply:
Removed